### PR TITLE
use unique_ptr in all `ut`

### PR DESCRIPTION
**Pull request by jackwener**
part of #663 .

use unique_ptr in all `ut`

### CODE REVIEW

File: tests/cppunit/compact_test.cc [hunk 1]
```diff
@@ -38,7 +38,7 @@ TEST(Compact, Filter) {
 
   int ret;
   std::string ns = "test_compact";
-  auto hash = new Redis::Hash(storage_.get(), ns);
+  auto hash = Util::MakeUnique<Redis::Hash>(Redis::Hash(storage_.get(), ns));
```
Comment (PragmaTwice) [inline tests/cppunit/compact_test.cc]:
```suggestion
  auto hash = Util::MakeUnique<Redis::Hash>(storage_.get(), ns);
```
And same as all of other changes.

File: tools/kvrocks2redis/parser.cc [hunk 2]
```diff
@@ -89,7 +89,7 @@ Status Parser::parseComplexKV(const Slice &ns_key, const Metadata &metadata) {
   read_options.iterate_upper_bound = &upper_bound;
   read_options.fill_cache = false;
 
-  auto iter = db_->NewIterator(read_options);
+  auto iter = std::unique_ptr<rocksdb::Iterator>(db_->NewIterator(read_options));
```
Comment (PragmaTwice) [inline tools/kvrocks2redis/parser.cc]:
```suggestion
  auto iter = DBUtil::UniqueIterator(db_, read_options);
```
and include "db_util.h" : )

Comment (jackwener) [reply tools/kvrocks2redis/parser.cc]: Thanks! ❤️

**Review by PragmaTwice — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Merged. Thanks for you contribution!

**Pull request merged**